source/Types.cpp

```diff
@@ -51,53 +51,53 @@ namespace Aws
             return aws_byte_cursor_from_array(array, len);
         }
 
-        Vector<uint8_t> Base64Decode(const String &decode)
+        Vector<uint8_t> Base64Decode(const String &decode) noexcept
         {
             ByteCursor toDecode = ByteCursorFromString(decode);
 
-            size_t allocation_size = 0;
+            size_t allocationSize = 0;
 
-            if (aws_base64_compute_decoded_len(&toDecode, &allocation_size) == AWS_OP_SUCCESS)
+            if (AWS_OP_SUCCESS != aws_base64_compute_decoded_len(&toDecode, &allocationSize))
             {
-                Vector<uint8_t> output(allocation_size, 0x00);
-                ByteBuf tempBuf = aws_byte_buf_from_array(output.data(), output.size());
-                tempBuf.len = 0;
-
-                if (aws_base64_decode(&toDecode, &tempBuf) == AWS_OP_SUCCESS)
-                {
-                    return output;
-                }
+                return {};
             }
+                
+            Vector<uint8_t> output(allocationSize, 0x00);
+            ByteBuf tempBuf = aws_byte_buf_from_empty_array(output.data(), output.capacity());
```

Comment:
.capacity is not size().

.size() will return allocationSize while capacity may return some value >=allocationSize
(of course typically it will return the size, but still capacity != size).

You are not calling resize / anything else that related to the actual capacity, so .size here is more correct.